### ISSUE TITLE
 X = proj_1 * proj_2 it may be wrong 

### ISSUE THREAD

**Issue by zyccccc** [luyao777/HBP-Pytorch#6]
def hbp(self,conv1,conv2):
        N = conv1.size()[0]
        proj_1 = self.bilinear_proj(conv1)
        proj_2 = self.bilinear_proj(conv2)
        assert(proj_1.size() == (N,8192,28,28))
        X = proj_1 * proj_2
        assert(X.size() == (N,8192,28,28))    
        X = torch.sum(X.view(X.size()[0],X.size()[1],-1),dim = 2)
        X = X.view(N, 8192)   
        X = torch.sqrt(X + 1e-5)
        X = torch.nn.functional.normalize(X)
        return X

in HBP , this line :X = proj_1 * proj_2. should be. self.bap = BAP(pool='GAP')?  
because this is just dot, it should be binear pooling ，You also have to multiply between the channels。
I think that's the right way to think about it，Thank you for the code 👍

**Comment by luyao777** [comment]
Thanks for your suggestion. Maybe you can check if this improves the results. :)
